src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1823,17 +1824,20 @@ abstract class ManagedStrategy(
         when (owner) {
             System.`in`, System.out, System.err -> return AnalysisSectionType.IGNORED
         }
-        val silentSection = getDefaultSilentSectionType(ownerName, methodName)
-            ?.ensure { it.isSilent() }
-        if (silentSection != null) {
-            return silentSection
-        }
+        val section = getAnalysisSectionFor(ownerName, methodName)
         userDefinedGuarantees?.forEach { guarantee ->
             if (guarantee.classPredicate(ownerName) && guarantee.methodPredicate(methodName)) {
                 return guarantee.type
             }
         }
-        return AnalysisSectionType.NORMAL
+        return if (section == AnalysisSectionType.IGNORED) AnalysisSectionType.SILENT else section
```

Comment:
So it turned out merging `IGNORED` section and instrumentation logic was not as easy as we thought initially. 
So let's postpone this change for now, create an issue ticket with the description of the problem and revisit it later. 

See my comment in the `AnalysisSection` file with the description of how I can proceed for now.